File: src/Domain/Models/Games/RequestsResults/RunTest.cs
using Domain.Entities.Challenges;
using Domain.Entities.Common;

namespace Domain.Models.Games.RequestsResults;

public class RunTestRequest
{
    public string Id { get; set; }
    public Solution Solution { get; set; }
    public TestPair Test { get; set; }
}
public class RunTestResult
{
    public string Id { get; set; }
    public string OutputError { get; set; }
    public TestResult Test { get; set; }
    public TestResult Validator { get; set; }
}

File: src/react-frontend/src/app/api/game/types/runTest.ts
import { TestPair } from 'app/types/models/challenge/testPair';
import { Solution } from 'app/types/models/general/solution';

export type TestResult = {
  note: string;
  status: string;
  buldStdout: string;
  buildStderr: string;
  buildExitCode?: number;
  buildTime: string;
  buildMemory?: number;
  buildResult: string;
  stdout: string;
  stderr: string;
  exitCode?: number;
  time: string;
  memory: string;
  connections: string;
  result: string;
};

export type RunTestRequest = {
  id: string;
  solution: Solution;
  test: TestPair;
};

export type RunTestResult = {
  id: string;
  outputError: string;
  test: TestResult;
  validator: TestResult;
};
